client/layout/masterbar/logged-out.jsx

```diff
@@ -31,7 +31,7 @@ const MasterbarLoggedOut = ( { title, sectionName, translate } ) => (
 			}
 
 			{ 'login' !== sectionName
-			?	<Item url={ config( 'login_url' ) }>
+			?	<Item url={ config( 'login_url' ) + '?redirect_to=' + encodeURIComponent( window.location.href ) }>
```

Comment:
Unless I'm missing something (maybe wrong config? as in production vs development), but if you're logged out in the desktop client, you have the option to create a site:
<img width="651" alt="screen shot 2017-04-17 at 18 40 33" src="https://cloud.githubusercontent.com/assets/3392497/25095771/7df2b418-239d-11e7-95bf-53fa2d48c816.png">

Clicking `Create account` switches the view to the one I know from `/start`:
<img width="880" alt="screen shot 2017-04-17 at 18 40 44" src="https://cloud.githubusercontent.com/assets/3392497/25095789/91082664-239d-11e7-9f12-3fae4c4243be.png">

In the top right you can see the `Log In` link I've added the `redirect_to` parameter to. Although, the `Sign Up` link shouldn't be there 😅 (since this view is exactly signup, at least in Calypso) - but that's not this PR's fault ;)

I've tested `wp-desktop` with this patch - as expected, the `redirect_to` parameter is not supported, but at least it doesn't seem to break anything ;)